server/ERNI.PBA.Server.Host/Controllers/RequestController.cs

```diff
@@ -302,6 +353,47 @@ public async Task<IActionResult> UpdateRequest([FromBody] UpdateRequestModel pay
             request.Amount = payload.Amount;
             request.Date = payload.Date.ToLocalTime();
 
+            var transactions = new[]
+            {
+                new Transaction
+                {
+                    RequestId = request.Id,
+                    BudgetId = budget.Id,
+                    UserId = currentUser.Id,
+                    Amount = payload.Amount
+                }
+            };
+            await _requestRepository.AddOrUpdateTransactions(request.Id, transactions);
+
+            await _unitOfWork.SaveChanges(cancellationToken);
+
+            return Ok();
+        }
+
+        [HttpPut("team")]
+        public async Task<IActionResult> UpdateTeamRequest([FromBody] UpdateRequestModel payload, CancellationToken cancellationToken)
```

Comment:
Now that you have a special controller for teambudgets, you can move also these methods